### DISCUSSION TITLE
Can I exclude document sections in Quarto when generating PDF output?

### DISCUSSION THREAD

**Discussion by IceManTBM3** [quarto-dev/quarto-cli#9615] [Q&A] [answered]
### Description

Below is my qmd code in the file index.qmd, there is a section "## Data Delivery, Help and FAQ " that I would like to omit, is there an easy way to exclude? Thanks in advance

---
title: RMBS Product Tear Sheet
# subtitle: ICE Sustainable Finance
author: "ICE Sustainable Finance"
# bibliography: references.bib
nocite: |
  @*
format:
  pdf:
    header-includes: |
      \titlehead{\includegraphics[width=6in]{img/ICE77x76.png}}
    toc: true
    toc-depth: 2
    number-sections: true
    number-depth: 2
    output-file: "RMBS-Tear"
    output-ext:  "pdf"
    cover-image: img/ICE77x76.png
    css: styles.css
    hyperrefoptions: draft
    colorlinks: true
    tbl-colwidths: false
    df-print: kable
    fig-align: left
    fig-pos: 'h'
#    linkcolor: RSSblue
#    urlcolor: RSSblue
    include-in-header: 
       - "styles.tex"
editor: source
---

```{r}
#| echo: false
#| eval: true
process_qmd <- function(file, fpath_in = "/", fpath_out = "/") {
  doc <- readLines(file)
  end_yaml <- which(doc == "---")[2]
  out_doc <- doc[seq(end_yaml+1, length(doc))]
  if (fpath_in != fpath_out) {
    out_doc <- stringr::str_replace_all(out_doc, fpath_in, fpath_out)
  }
  res <- knitr::knit_child(text = out_doc, quiet = TRUE, options = list(eval = FALSE, echo = TRUE))
  return(res)
}
```

# Mission/Overview

```{r}
#| output: asis
#| echo: false
#| eval: true
#| message: false
idx <- process_qmd("index.qmd")
cat(unlist(idx), sep = '\n')
```

# Disclaimer

```{r}
#| output: asis
#| echo: false
#| eval: true
#| message: false
idx <- process_qmd("disclaimer.qmd")
cat(unlist(idx), sep = '\n')
```

**Comment by mcanouil** [comment, marked as answer]
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.


This being said you likely are interested in "conditional content" as described in <https://quarto.org/docs/authoring/conditional.html>.

**Reply by IceManTBM3** [reply]
yeah that pasted in funny

your solution works, that is exactly what I was looking for

**Reply by mcanouil** [reply]
@IceManTBM3 We kindly ask that you fulfill the reformatting request. Without a properly formatted message, future users may find it impossible to read, understand, or use the discussion. Thank you for your cooperation!

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Comment by JackGraymer** [comment]
Same question. 
Writing a README.md for a project, but also outputting it in pdf to then share on a printed version.
There is a section in a expandable section (HTML `<details></details>`. On github and HTML this works perfectly, but on PDF it is expanded, which makes sense, as it is content that otherwise would be lost on PDF.

This is exactly what I would like, how can I leave some markdown out for the PDF version, in this case, I dont want the hidden `<details>` section to be expanded and printed. Is there a way to hid this on PDF?

<details>
<summary>Click to expand</summary>
Example
This part should not be included in final PDF
</details>

Thanks

**Reply by mcanouil** [reply]
Use conditional content as documented on quarto.org.

Note that you are mistaken about what is happening with "pdf", i.e., the content is not expanded as you seem to think.
You have HTML markups. Those have no meanings in LaTeX so they are discarded completed by Pandoc.

**Reply by JackGraymer** [reply]
Thanks, it works as intended but not for .md files.
The idea was to use it on a README.md, that can then output html with the `details` collapsed, and that is not there on HTML.
It works, but the md file shows on github the conditional content code:
```
::: {.content-visible when-format="html"}

Will only appear in HTML.

:::```

**Reply by mcanouil** [reply]
Care to share a small fully reproducible example?

Note also that `README.*` are excluded from rendering in Quarto projects.
Quarto will only render them if explicitly asked for.

> It works, but the md file shows on github the conditional content code:

Of course, why would GitHub not show what you put in your README?

I suggest you create a document "source-readme.qmd", then `quarto render source-readme.qmd --to all`:

````qmd
---
format:
  gfm:
    output-file: README
  html: default
---

whatever content.

::: {.content-visible when-format="html"}

But not this one for README.md

:::
````

**Reply by JackGraymer** [reply]
It does work. My idea is that I am writing the document purely on markdown as the readme.md file of a repository.
Then that is rendered as html and pdf. Conditional formatting works in both cases as expected, just that the readme file shows the code, as you pointed out, github does show everything in the file, including the conditional block, which is not very clean on documentation.

Thank you, I think this is clear, solution is to write it as `.qmd` and render it to all other formats: markdown, html and pdf

**Reply by mcanouil** [reply]
> Thank you, I think this is clear, solution is to write it as .qmd and render it to all other formats: markdown, html and pdf

That's only one solution, but I do think that it's the cleanest.

**Comment by IceManTBM3** [comment]
Use ::: {.content-visible unless-format="pdf"}

**Reply by mcanouil** [reply]
@IceManTBM3 your message is illegible. Please don't use email to reply to GitHub or ensure they will be legible. If you really want to reply with email, do not include previous replies. This is not a mail exchange.
Thank you for your cooperation.
